### ISSUE TITLE
[Quick 7] Undocumented Breaking change from 6 --> 7

### ISSUE THREAD

**Issue by homestar9** [coldbox-modules/quick#234]
I discovered a breaking change from Quick 6 that I didn't see covered in the docs that might be worth adding.  When dealing with casts when the database has null values.  I created a little test using `BooleanCast@quick` as an example.

If you have your entity set up like this:
```
component 
    extends="quick.models.BaseEntity"
    accessors="true"
    table="test"
{

    property name="id";
    property name="name" sqltype="varchar";
    property name="active" sqltype="bit" casts="BooleanCast@quick"; // <--- boolean cast

}
```

Return an entity with a null database value in the `active` field.  
```
var entity = wirebox.getInstance( "TestEntity" ).find( 1 );
debug( entity.getMemento() );  // testbox debug
```

Quick 5 will output:
![image](https://github.com/coldbox-modules/quick/assets/5341142/cd9d8a38-6ca7-4dd7-a486-5a364aa0be45)

Quick 6 will output:
![image](https://github.com/coldbox-modules/quick/assets/5341142/11528f84-bca2-4219-b184-3116418ce51c)

Quick 7 will output:
![image](https://github.com/coldbox-modules/quick/assets/5341142/ad09628e-a85a-4d41-94f7-5aa653ed2765)

The same is true for new (unpersisted) entities as well:
Quick 5 and 6:
![image](https://github.com/coldbox-modules/quick/assets/5341142/077a4b81-7a43-4c43-bb20-74003f1ec4e3)

Quick 7:
![image](https://github.com/coldbox-modules/quick/assets/5341142/43346e1a-e209-48d3-b70e-1010237bf0e2)

I don't know what the proper behavior is, but it's a notable difference people upgrading might want to know about. Personally, I preferred the old behavior where a null value was cast to `false`.  Either way, updating to Quick 7 caused many of my tests to fail.

**Comment by bdw429s** [comment]
This issue has been mentioned on **Ortus Solutions Community**. There might be relevant details there:

https://community.ortussolutions.com/t/quick-7-undocumented-breaking-change-from-6-to-7/9731/1

**Comment by elpete** [comment]
This was intentional, and was actually fixed [in 7.0.1 as a bug fix.](https://quick.ortusbooks.com/whats-new#7.0.1) The new behavior is the desired behavior since a `NULL` value does not necessarily mean `false`. It could mean not set yet. If you want a default value, you should be able to set `default="false"` on the property.
